*** orig/exim4.conf.orig	2017-01-13 23:50:35.684545721 +0100
--- patch/stsbl-iserv-mail-send/exim4.conf	2017-01-14 18:05:47.024241101 +0100
***************
*** 160,165 ****
--- 160,182 ----
        message = You are not allowed to send e-mail to other servers.\n.\n\
          Sie dürfen keine E-Mails an andere Server schicken.
  
+     # Deny internal sending of mails to groups without mail_int flag
+     deny
+       authenticated = *
+       condition = ${lookup pgsql{ \
+          SELECT 1 FROM groups g WHERE g.act = '${quote_pgsql:$local_part}' \
+          AND NOT EXISTS (SELECT 1 FROM groups_flag f WHERE f.act=g.act AND \
+          f.flag='mail_int') AND NOT EXISTS (SELECT 1 FROM users_priv p \
+          WHERE p.act = '${quote_pgsql:$authenticated_id}' AND p.privilege = \
+          'mail_all_grps' LIMIT 1)}}
+       # Admins must always reachable
+       !local_parts = admins
+       # only apply for local domains
+       domains = +local_domains
+         message = This recipient is not allowed to receive e-mail from other \
+         internal e-mail accounts.\n\nDieser Empfänger darf keine E-Mails von \
+ 	anderen internen E-Mail-Konten empfangen.
+ 
      # Variable acl_m_ext_rcpt setzen, wenn mindestens ein externer Empfänger
      warn
        !domains = +local_domains
